skops/hub_utils/_hf_hub.py

```diff
@@ -415,19 +418,64 @@ def init(
     ):
         requirements.append("scikit-learn-intelex")
 
+    # model parameter can be either a path or a model object
     try:
-        shutil.copy2(src=model, dst=dst)
-
-        model_name = model.name
-        _create_config(
-            model_path=model_name,
-            requirements=requirements,
-            dst=dst,
-            task=task,
-            data=data,
-            model_format=model_format,
-            use_intelex=use_intelex,
-        )
+        if isinstance(model, (str, Path)):
+            model = _check_model_file(model)
+            shutil.copy2(src=model, dst=dst)
+
+            model_name = model.name
+
+            _create_config(
+                model_path=model_name,
+                requirements=requirements,
+                dst=dst,
+                task=task,
+                data=data,
+                model_format=model_format,
+                use_intelex=use_intelex,
+            )
+
+            # load model from file
+            model = _load_model(model, trusted=True)
+            model_card = card.Card(model, metadata=card.metadata_from_config(dst))
+            model_card.save(dst / "README.md")
+        elif isinstance(model, BaseEstimator):
+            # if it is a model object and its format is set to auto, choose skops by default
+            if model_format == "auto":
+                model_format = "skops"
+            elif model_format in ["pkl", "pickle", "joblib"]:
+                model_format = "pickle"
```

Comment:
We need an `else` clause below because if the user passes `model_format="skosp"` or another typo, that would just be accepted as is, even though it is invalid. Then further below, the model would not be saved without any indication that something went wrong.

Essentially, the model format checking logic should be the same as inside `_create_config`. And since checking the model format is new performed in `init`, we should also be able to remove it from `_create_config` completely.

Reply:
If so, I can remove that logic from `_create_config` and have it just in `init`. I actually started this PR by looking at `_create_config`. Let me know if you would like to make this change now.

Reply:
Yes, I think that change should be good.